src/core/searcher.rs

```diff
@@ -11,8 +12,52 @@ use crate::store::StoreReader;
 use crate::DocAddress;
 use crate::Index;
 
+use std::collections::BTreeMap;
+use std::ops::Deref;
+use std::sync::Arc;
+use std::sync::Weak;
 use std::{fmt, io};
 
+/// Identifies the index generation accessed by a [Searcher].
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct SearcherIndexGeneration(BTreeMap<SegmentId, u32>);
+
+impl SearcherIndexGeneration {
+    pub(crate) fn from_segment_readers(readers: &[SegmentReader]) -> Self {
+        let mut segment_id_to_del_docs = BTreeMap::new();
+        for reader in readers {
+            segment_id_to_del_docs.insert(reader.segment_id(), reader.num_deleted_docs());
```

Comment:
Ah interesting. Technically this is correct because we cannot undelete a message, but I'd rather we used the `delete_opstamp` because this is what we use to identify a segment version in the code.

Reply:
I think the change is simple, rename the function `from_segment_metas` and pass it SegmentMetas.

The call site already has them.

Reply:
> I'm ok to consider refactoring tantivy to num_deleted_docs though if we see some cool benefits.

Hmm, I think the only benefit is that it is a bit more meaningful in interpreting, opstamp feels opaque. Also `Option<u64>` vs `u32` - there is no need for optionality.

I changed it to the delete opstamp for consistency.

Reply:
Yes that's interesting. I'll open a ticket.